### PR TITLE
Vicky/authorize request

### PR DESCRIPTION
**Pull request by vrodriguero**
This **PR** includes the following changes:

### Create a new authorize request
This file includes the following functions:
- Getter and Setter for `OrderID`
- Getter for `data`
- Getter for `endpoint` is `/v2/checkout/orders/{id}/authorize` (see [PayPal](https://developer.paypal.com/docs/api/orders/v2/#orders_authorize) docs)
- A method that returns a new **orders response** which takes the `data` and the `status code` as arguments

### Add a method named authorize in the Checkout Gateway
- The method takes an array of `parameters` as an argument
- The method will return a new request using the **Authorize request** created. 

### Create examples of Authorization Failure and Success
- **success:** create a mockup of a response from the example in [PayPal](https://developer.paypal.com/docs/api/orders/v2/#orders_authorize) docs
- **failure**: create a mockup of a response from the example in the **complete purchase** Mockup response

### Added tests to check the implementation
- **AuthorizeRequestTest**: test `data` received and `endpoint`
- **ResponseTest**: test Authorization `failure` and `success` response examples
- **CheckoutGateway**: add tests for Authorization `failure` and `success`

#### _Please Note_:
The difference between **ResponseTest** and **CheckoutGateway** is that:
  -  **ResponseTest** uses the method that **gets** the `MockHttpResponse` for a client by the file name 
  - **CheckoutGateway** uses the method that **sets** the `MockHttpResponse` from the mockup on the next client request.

### CODE REVIEW

File: src/Message/AbstractRequest.php [hunk 1]
```diff
@@ -93,23 +86,6 @@ public function setSecret($value): AbstractRequest
...
-    }
-
-    /**
-     * @param $value
-     * @return AbstractRequest
-     */
-    public function setPayerId($value): AbstractRequest
-    {
-        return $this->setParameter('payerId', $value);
-    }
```
Comment (stevethomas) [inline src/Message/AbstractRequest.php]: @vrodriguero why did you remove these from AbstractRequest?

Comment (vrodriguero) [reply src/Message/AbstractRequest.php]: Oh, I am sorry!
I must have deleted by mistake, I remember that when we were discussing the authorize method, we would not need the payer_id on the getData() method. I must have deleted at that time, as I thought that we would only include in the authorize.

**Review by stevethomas — APPROVED** [review]
